backend/indexing/run_job.py

```diff
@@ -37,14 +40,26 @@
 def run_task(task: IndexTask) -> None:
     '''Run an IndexTask'''
     task_id = f'{task.__class__.__name__} #{task.pk}' # e.g. "CreateIndexTask #1"
+
+    try:
+        raise_if_aborted(task)
```

Comment:
This seems convoluted. Will the `raise` not occur in the natural flow of indexing (when the task is aborted)? Now it seems like i you forget to ask specifically for a `raise` check, the program fails silently.